src/Build/BuildCheck/Infrastructure/ConfigurationProvider.cs

```diff
@@ -11,64 +11,27 @@
 using System.Text.Json;
 using Microsoft.Build.Experimental.BuildCheck;
 using System.Configuration;
+using Microsoft.Build.BuildCheck.Infrastructure.EditorConfig;
 
 namespace Microsoft.Build.BuildCheck.Infrastructure;
 
 
 // TODO: https://github.com/dotnet/msbuild/issues/9628
-//  Let's flip form statics to instance, with exposed interface (so that we can easily swap implementations)
-internal static class ConfigurationProvider
+internal class ConfigurationProvider
 {
+    private EditorConfigParser s_editorConfigParser = new EditorConfigParser();
+
     // TODO: This module should have a mechanism for removing unneeded configurations
     //  (disabled rules and analyzers that need to run in different node)
-    private static readonly Dictionary<string, BuildAnalyzerConfiguration> _editorConfig = LoadConfiguration();
-
-    // This is just a testing implementation for quicker unblock of testing.
-    // Real implementation will use .editorconfig file.
-    // Sample json:
-    /////*lang=json,strict*/
-    ////"""
-    ////    {
-    ////        "ABC123": {
-    ////            "IsEnabled": true,
-    ////            "Severity": "Info"
-    ////        },
-    ////        "COND0543": {
-    ////            "IsEnabled": false,
-    ////            "Severity": "Error",
-    ////    		"EvaluationAnalysisScope": "AnalyzedProjectOnly",
-    ////    		"CustomSwitch": "QWERTY"
-    ////        },
-    ////        "BLA": {
-    ////            "IsEnabled": false
-    ////        }
-    ////    }
-    ////    """
-    //
-    // Plus there will need to be a mechanism of distinguishing different configs in different folders
-    //  - e.g. - what to do if we analyze two projects (not sharing output path) and they have different .editorconfig files?
-    private static Dictionary<string, BuildAnalyzerConfiguration> LoadConfiguration()
-    {
-        const string configFileName = "editorconfig.json";
-        string configPath = configFileName;
-
-        if (!File.Exists(configPath))
-        {
-            // TODO: pass the current project path
-            var dir = Environment.CurrentDirectory;
-            configPath = Path.Combine(dir, configFileName);
+    private readonly Dictionary<string, BuildAnalyzerConfiguration> _editorConfig = new Dictionary<string, BuildAnalyzerConfiguration>();
 
-            if (!File.Exists(configPath))
-            {
-                return new Dictionary<string, BuildAnalyzerConfiguration>();
-            }
-        }
+    private readonly Dictionary<string, CustomConfigurationData> _customConfigurationData = new Dictionary<string, CustomConfigurationData>();
```

Comment:
I don't fully understand the purpose of this field. Can you please add a comment? It looks like a cache but it's not used outside of the validity check.

Reply:
The purpose of the _customConfigurationData is to save the first custom data for specific rule. 
Since there could be different configurations based on the editorconfig and project file, we want to make sure that the custom data between different projects are the same.